zebra-network/src/meta_addr.rs

```diff
@@ -688,7 +712,11 @@ impl MetaAddrChange {
                 last_failure: None,
                 last_connection_state: self.peer_addr_state(),
             }),
-            UpdateAttempt { .. } | UpdateResponded { .. } | UpdateFailed { .. } => None,
+
+            UpdateAttempt { .. }
+            | UpdateResponded { .. }
+            | UpdateFailed { .. }
+            | AddPeer { .. } => None,
```

Comment:
I think this is the cause of the test failures. `MetaAddrChange::into_new_meta_addr` should return `Some(MetaAddr { .. })` for `MetaAddrChange::AddPeer`.

However, to that you will need to either have `services` be optional (I opened #2976 for this, but I'm not sure if it will be reviewed and merged in time), or just assume initial peers have `PeerServices::NODE_NETWORK`.

Reply:
I don't mind which alternative we use here.

PR #2976 is almost ready to merge, feel free to approve it once the test coverage issue is fixed:
https://github.com/ZcashFoundation/zebra/pull/2976#pullrequestreview-793787887

Reply:
I'd go with @jvff's first suggestion since he made the effort to implement it.